### PR TITLE
ft: local follows (at the least) working, attempt to send posts to remote followers

### PR DESCRIPTION
**Pull request by Mattwmaster58**
in this PR:
 - attempt to send posts to endpoint of followers, log if/when it fails
 - refactor follow mechanisms: remove fancy sqla tables, I couldn't understand or more importantly debug them, i do not think they were necessary
 - change our default object id creation thing: I found it annoying 2 ids could be similar at the start and end of the string, varying only in the middle so i just swapped it out for something else entirely
missing: inbox follow requests, need to re-factor for it

### CODE REVIEW

File: api/user/followers/followers.py [hunk 1]
```diff
@@ -127,30 +124,49 @@ def remove_follower(author_id: str, foreign_author_id: str):
...
+    if not followed_obj:
+        return {"message": f"{followed_id=} doesn't exist!"}, 404
+
+    follower_to_add = Author.query.filter_by(id=follower_id).first()
+    params = {"followed_id": followed_id, "follower_id": follower_id}
+    if follower_to_add:
+        if LocalFollower.query.filter(
+            and_(LocalFollower.followed_id == followed_id, LocalFollower.follower_id == follower_id)
+        ).count():
+            return {"success": False, "message": "You are already following this user!"}
```
Comment (shehraj123) [inline api/user/followers/followers.py:142]: can this be a 409

File: api/user/followers/followers.py [hunk 2]
```diff
@@ -101,9 +98,9 @@ def followers_count(author_id: str):
 @basic_auth.required
 def following_count(author_id: str):
     """Get the count for the number of people author is following"""
-    following = Author.query.filter_by(id=author_id).join(follows_table, follows_table.c.follower_id == Author.id).all()
-    non_local_following = NonLocalFollower.query.filter_by(follower_id=author_id).all()
-    return {"count": len(following) + len(non_local_following)}, 200
+    following = LocalFollower.query.filter_by(follower_id=author_id).count()
+    non_local_following = NonLocalFollower.query.filter_by(follower_id=author_id).count()
```
Comment (shehraj123) [inline api/user/followers/followers.py:102]: NonLocalFollower table has followed_id column as foreignkey from author.id and Author table doesnt have foreign authors in it. I think we need another table for NonLocalFollowing

Comment (shehraj123) [reply api/user/followers/followers.py:102]: Check model.py for followers

File: api/user/followers/model.py [hunk 3]
```diff
@@ -2,14 +2,15 @@
...
-follows_table = db.Table(
-    "follows",
-    db.Column("follower_id", db.Text, db.ForeignKey("author.id"), primary_key=True),
-    db.Column("followed_id", db.Text, db.ForeignKey("author.id"), primary_key=True),
-)
+
+class LocalFollower(db.Model):
+    follower_id: str = db.Column("follower_id", db.Text, db.ForeignKey("author.id"), primary_key=True)
+    followed_id: str = db.Column("followed_id", db.Text, db.ForeignKey("author.id"), primary_key=True)
+    approved: bool = db.Column("approved", db.Boolean, nullable=False, default=False)
```
Comment (shehraj123) [inline api/user/followers/model.py:10]:
```suggestion
class NonLocalFollowing(db.Model):
> followed_id: str = db.Column(db.Text, primary_key=True, nullable=False)
>follower_id: str = db.Column(db.Text, db.ForeignKey("author.id"), primary_key=True)
>approved: bool = db.Column(db.Boolean, nullable=False, default=False)

```

File: api/user/posts/posts.py [hunk 4]
```diff
@@ -580,6 +585,27 @@ def fanout_to_local_inbox(post: Post, author: str = None):
         pass
 
 
+def fanout_to_foreign_inbox(post, author_id):
+    all_foreign = NonLocalAuthor.query.filter_by(followed_id=author_id).all()
```
Comment (shehraj123) [inline api/user/posts/posts.py:589]:
```suggestion
    all_foreign = NonLocalFollower.query.filter_by(followed_id=author_id).all()
```

File: api/utils.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 import re
 import time
 from dataclasses import asdict, dataclass
+from string import ascii_lowercase
```
Comment (shehraj123) [inline api/utils.py:8]:
```suggestion
import uuid
```

**Review by Manpreet-Singh001 — COMMENTED** [review]
Can you please address that this won't return a list of followers for the /followers endpoint due to the tables being removed, other than that it looks good to go

**Review by Manpreet-Singh001 — APPROVED** [review]

**Pull request merged**